molecule/molecule-runtime/src/commonMain/kotlin/app/cash/molecule/GatedFrameClock.kt

```diff
@@ -0,0 +1,42 @@
+package app.cash.molecule
+
+import androidx.compose.runtime.BroadcastFrameClock
+import androidx.compose.runtime.MonotonicFrameClock
+import kotlin.time.ExperimentalTime
+import kotlin.time.TimeSource
+
+/**
+ * A [MonotonicFrameClock] that is either running, or not.
+ *
+ * While running, any request for a frame immediately succeeds. If stopped, requests for a frame wait until
+ * the clock is set to run again.
+ */
+@OptIn(ExperimentalTime::class)
+internal class GatedFrameClock : MonotonicFrameClock {
+  private val start = TimeSource.Monotonic.markNow()
+
+  var isRunning: Boolean = true
+    set(value) {
+      val started = value && !field
+      field = value
+      if (started) {
+        sendFrame()
+      }
+    }
+
+  private fun sendFrame() {
+    clock.sendFrame(start.elapsedNow().inWholeNanoseconds)
```

Comment:
Is this time needed for anything? The existing testing code just sends `0` because it doesn't matter.

Reply:
Not that I know of. Is there any downside to providing a meaningful value?

Reply:
The "downside" is that this is doing work that it might not need to do 🤷 I don't know enough about the internals to say whether this is truly meaningless, meaningless outside of UI, or meaningful in some subtle way.

Reply:
In JVM land, this would no doubt turn into a `System.nanoTime()` in the `actual` implementation. So I can't imagine it's too expensive.